server/src/main/java/org/apache/druid/server/coordinator/loading/StrategicSegmentAssigner.java

```diff
@@ -431,7 +433,7 @@ private int dropReplicas(
       Iterator<ServerHolder> serverIterator =
           (useRoundRobinAssignment || eligibleLiveServers.size() >= remainingNumToDrop)
```

Comment:
Should this be`eligibleLiveServers.size() <= remainingNumToDrop` ?

Reply:
Yes, I think so. Thanks for catching this!